Missing libnss3.so after updating openjdk:8-jdk-alpine image

Hello,

I have a docker file that builds openjdk:8-jdk-alpine based images. When updating the image internally (from the docker file) using `apk update && apk upgrade ` , and running my (spring-boot) app, it throws `java.io.IOException: Error loading shared library libnss3.so: No such file or directory`. However; after removing the `apk update && apk upgrade ` and building the image again, the app runs smoothly. 

I have logged into my images in both cases, and made sure that the file wasn't there on the first case, and was there after removing `apk update && apk upgrade `. In the later case, it was located in `/usr/lib/`

Is this a bug, or can you please inform me of what I might be missing ?

We've also just been hit with this. We execute an `apk update && apk upgrade` during each build of our image, and today we've started getting the following error during start up of our Tomcat application:

```
Caused by: java.io.FileNotFoundException: /usr/lib/libnss3.so
	at sun.security.pkcs11.Secmod.initialize(Secmod.java:193)
	at sun.security.pkcs11.SunPKCS11.<init>(SunPKCS11.java:218)
	... 100 more
```

The root cause appears to be the PostgreSQL driver establishing an SSL connection to an RDS endpoint.

Here's the output from our `apk upgrade`:

```
13:17:57  (1/7) Upgrading libcrypto1.1 (1.1.1a-r1 -> 1.1.1b-r1)
13:17:57  (2/7) Upgrading libssl1.1 (1.1.1a-r1 -> 1.1.1b-r1)
13:17:57  (3/7) Upgrading openjdk8-jre-base (8.191.12-r0 -> 8.201.08-r0)
13:17:57  (4/7) Upgrading openjdk8-jre (8.191.12-r0 -> 8.201.08-r0)
13:17:57  (5/7) Purging nss (3.41-r0)
13:17:57  (6/7) Purging nspr (4.20-r0)
13:17:57  (7/7) Purging sqlite-libs (3.26.0-r3)
```

I'm not sure which of the upgraded packages is causing APK to purge `nss`, but following up with `apk add --no-cache nss` seems to bring it back.

--- a/8/jdk/alpine/Dockerfile
+++ b/8/jdk/alpine/Dockerfile
@@ -31,13 +31,16 @@ RUN { \
 ENV JAVA_HOME /usr/lib/jvm/java-1.8-openjdk
 ENV PATH $PATH:/usr/lib/jvm/java-1.8-openjdk/jre/bin:/usr/lib/jvm/java-1.8-openjdk/bin
 
-ENV JAVA_VERSION 8u191
-ENV JAVA_ALPINE_VERSION 8.191.12-r0
+ENV JAVA_VERSION 8u201
+ENV JAVA_ALPINE_VERSION 8.201.08-r0
 
 RUN set -x \
 	&& apk add --no-cache \
 		openjdk8="$JAVA_ALPINE_VERSION" \
 	&& [ "$JAVA_HOME" = "$(docker-java-home)" ]
+
+# https://bugs.alpinelinux.org/issues/10126
+RUN apk add --no-cache so:libnss3.so
 
 # If you're reading this and have any feedback on how this image could be
 # improved, please open an issue or a pull request so we can discuss it!

--- a/8/jre/alpine/Dockerfile
+++ b/8/jre/alpine/Dockerfile
@@ -31,13 +31,16 @@ RUN { \
 ENV JAVA_HOME /usr/lib/jvm/java-1.8-openjdk/jre
 ENV PATH $PATH:/usr/lib/jvm/java-1.8-openjdk/jre/bin:/usr/lib/jvm/java-1.8-openjdk/bin
 
-ENV JAVA_VERSION 8u191
-ENV JAVA_ALPINE_VERSION 8.191.12-r0
+ENV JAVA_VERSION 8u201
+ENV JAVA_ALPINE_VERSION 8.201.08-r0
 
 RUN set -x \
 	&& apk add --no-cache \
 		openjdk8-jre="$JAVA_ALPINE_VERSION" \
 	&& [ "$JAVA_HOME" = "$(docker-java-home)" ]
+
+# https://bugs.alpinelinux.org/issues/10126
+RUN apk add --no-cache so:libnss3.so
 
 # If you're reading this and have any feedback on how this image could be
 # improved, please open an issue or a pull request so we can discuss it!

--- a/update.sh
+++ b/update.sh
@@ -355,6 +355,14 @@ RUN set -x \\
 	&& [ "\$JAVA_HOME" = "\$(docker-java-home)" ]
 EOD
 
+			if [ "$alpinePackageVersion" = '8.201.08-r0' ]; then
+				cat >> "$dir/alpine/Dockerfile" <<-EOD
+
+					# https://bugs.alpinelinux.org/issues/10126
+					RUN apk add --no-cache so:libnss3.so
+				EOD
+			fi
+
 			template-contribute-footer >> "$dir/alpine/Dockerfile"
 		elif [ -d "$dir/alpine" ]; then
 			downloadUrl="$(jdk-java-net-download-url "$javaVersion" '_linux-x64-musl_bin.tar.gz')"

